### PR TITLE
[v4] Update React - preparations

### PR DESCRIPTION
**Pull request by AndrewMusgrave**
### WHY are these changes introduced?

Unfortunately we cannot continue updating our react components to prepare for react 16.8.* until we update to 16.8.* 😅 [Enzyme doesn't support the new context](https://github.com/airbnb/enzyme/issues/1908) at our current version, a bump in the version require a bump in other packages require a bump in react 

### WHAT is this pull request doing?

Updating react / enzyme

### CODE REVIEW

File: config/typescript/prop-types.d.ts [hunk 1]
```diff
@@ -1,5 +0,0 @@
-// https://github.com/Microsoft/TypeScript/issues/24599
-declare module 'prop-types' {
```
Comment (AndrewMusgrave) [inline config/typescript/prop-types.d.ts]: This was causing namespace issues

File: src/components/AppProvider/tests/AppProvider.test.tsx [hunk 2]
```diff
@@ -1,57 +1,9 @@
 import * as React from 'react';
-import TestUtils from 'react-dom/test-utils';
 import {mount} from 'enzyme';
-import {createThemeContext} from '../../ThemeProvider';
-import {StickyManager, createAppProviderContext} from '../utilities';
 import {polarisAppProviderContextTypes} from '../types';
 import AppProvider from '../AppProvider';
 
 describe('<AppProvider />', () => {
-  it('passes i18n and withComponent properties to context', () => {
```
Comment (AndrewMusgrave) [inline src/components/AppProvider/tests/AppProvider.test.tsx]: This test will be updated and improved when we switch `AppProvider` to the new context API

File: src/components/Modal/tests/Modal.test.tsx [hunk 3]
```diff
@@ -312,18 +312,20 @@ describe('<Modal>', () => {
   });
 
   describe('with app bridge', () => {
+    let AppBridgeModalCreate: jest.SpyInstance;
```
Comment (AndrewMusgrave) [inline src/components/Modal/tests/Modal.test.tsx]: The next few files are using jest API to spy on functions rather than overwrite the function with a spy

**Comment by AndrewMusgrave** [comment]
@danrosenthal I was planning on merging this into `v4` since it has a ✅and every feature merged into this was ✅. Let me know if I should wait for a review from yourself/Tim 😄

**Comment by danrosenthal** [comment]
> @danrosenthal I was planning on merging this into v4 since it has a ✅and every feature merged into this was ✅. Let me know if I should wait for a review from yourself/Tim 😄

No need, was just adding us there so we could follow along.

**Pull request merged**